packages/app-headless-cms/src/admin/components/ContentEntries/Filters/FilterByStatus.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useMemo } from "react";
+
+import { useBind } from "@webiny/form";
+import { plugins } from "@webiny/plugins";
+import { Select } from "@webiny/ui/Select";
+
+import { DropdownContainer } from "./styles";
+
+import { CmsEntryFilterStatusPlugin } from "@webiny/app-headless-cms-common/types";
+
+export const FilterByStatus: React.FC = () => {
+    const bind = useBind({
+        name: "status",
+        defaultValue: "",
+        beforeChange(value, cb) {
+            cb(value === "all" ? undefined : value);
+        }
+    });
+
+    const filterStatusPlugins = useMemo(() => {
```

Comment:
You can build the options list immediately in this `useMemo()`. No need to memoize the plugins and then on each render add option items from the plugins to the options array.